### PR TITLE
PanelEditor: add data provider when switching from non data panel

### LINKED ISSUE #94206: PanelEdit: critical error when switching from panel with no tabs
error happens in panel edit when switching from a saved panel that does not have panel tabs (like Queries, Trasformations etc)  to a panel that has tabs. Text and Alert list panels are causing this issue.

https://github.com/user-attachments/assets/70a4f4b5-a326-4c96-964d-96acb855ff19

### PR DESCRIPTION
**Pull request by Sergej-Vlasov**
add data provider when switching from a non data panel
 
Fixes
#94206

### CODE REVIEW

File: public/app/features/dashboard-scene/panel-edit/PanelOptionsPane.tsx [hunk 1]
```diff
@@ -76,6 +80,25 @@ export class PanelOptionsPane extends SceneObjectBase<PanelOptionsPaneState> {
...
+          $data: new SceneQueryRunner({
+            datasource: {
+              uid: ds,
+            },
+            queries: [{ refId: 'A' }],
+          }),
+          transformations: [],
+        }),
+      });
+    }
```
Comment (torkelo) [inline public/app/features/dashboard-scene/panel-edit/PanelOptionsPane.tsx:100]: this feels like the wrong place to fix this bug, the bug is we are rendering the data plane when we should not

Comment (torkelo) [reply public/app/features/dashboard-scene/panel-edit/PanelOptionsPane.tsx:100]: thinking that we either add something to onChangePanelPlugin to clear data plane when the plugin is a non data plugin.

Comment (Sergej-Vlasov) [reply public/app/features/dashboard-scene/panel-edit/PanelOptionsPane.tsx:100]: I wanted to add this logic in PanelEditor in [`_updateDataPane`](https://github.com/grafana/grafana/blob/4623a6471bf77036ea3f0f17b0ff4e94272bb01a/public/app/features/dashboard-scene/panel-edit/PanelEditor.tsx#L183). Would it be better to add `$data` and clear `$data` there?

**Comment by mdvictor** [comment]
Tested and LGTM! Can we please add some tests to cover this scenario?

**Review by mdvictor — APPROVED** [review]

**Comment by Sergej-Vlasov** [comment]
> Tested and LGTM! Can we please add some tests to cover this scenario?

I added check for initial data provider state, but I could not test the actual plugin type switch. I will make a note and figure it out a bit later

**Pull request merged**